Kernel/Arch/x86/common/InterruptManagement.cpp

```diff
@@ -143,15 +143,7 @@ UNMAP_AFTER_INIT void InterruptManagement::switch_to_pic_mode()
     m_interrupt_controllers.append(adopt_lock_ref(*new PIC()));
     SpuriousInterruptHandler::initialize(7);
     SpuriousInterruptHandler::initialize(15);
-    for (auto& irq_controller : m_interrupt_controllers) {
-        VERIFY(irq_controller);
-        if (irq_controller->type() == IRQControllerType::i82093AA) {
-            irq_controller->hard_disable();
```

Comment:
Don't we need to hard disable the IOAPIC in case of switching to use the PIC?

Reply:
No, if we got to this function we don't have an IOAPIC in the `m_interrupt_controllers` list, so this should never happen.

Reply:
Ah, makes sense. Mind adding a sentence about that to the commit message?

Reply:
Sure

Reply:
Please add `VERIFY(m_interrupt_controllers.size() == 1)` to ensure this stays true.

Reply:
Done & done